Hometask2.playground/Contents.swift

```diff
@@ -0,0 +1,178 @@
+
+
+func digitsSum(_ firstNumber: Int, plus secondNumber: Int) -> Int {
+    return firstNumber + secondNumber
+}
+
+func digitsSum(_ number: Double, plus: Double) -> Int {
+    return Int(number + plus)
+}
+
+func digitsSum(_ number: Int, plus: Int) -> Double {
+    return Double(number + plus)
+}
+
+func digitsSum(_ number: Double, plus: Double) -> Double {
+    return number + plus
+}
+
+//Subtraction implementation via overloading
+
+let theFirstSubtraction: Int     = digitsSubtraction(2, minus: 2)
+let theSecondSubtraction: Double = digitsSubtraction(4, minus: 8)
+let theThirdSubtraction: Int     = digitsSubtraction(5.0, minus: 8.0)
+let theFourthSubtraction: Double = digitsSubtraction(5.5, minus: 5.6)
+
+func digitsSubtraction(_ firstNumber: Int, minus secondNumber: Int) -> Int {
+    return firstNumber - secondNumber
+}
+
+func digitsSubtraction(_ firstNumber: Double, minus secondNumber: Double) -> Int {
+    return Int(firstNumber - secondNumber)
+}
+
+func digitsSubtraction(_ firstNumber: Int, minus secondNumber: Int) -> Double {
+    return Double(firstNumber - secondNumber)
+}
+
+func digitsSubtraction(_ firstNumber: Double, minus secondNumber: Double) -> Double {
+    return firstNumber - secondNumber
+}
+
+//Multiplication implementation via overloading
+
+let theFirstMultiply: Int     = digitsMultiplication(2, by: 2)
+let theSecondMultiply: Double = digitsMultiplication(4, by: 8)
+let theThirdMultiply: Int     = digitsMultiplication(5.0, by: 8.0)
+let theFourthMultiply: Double = digitsMultiplication(5.5, by: 5.6)
+
+func digitsMultiplication(_ firstNumber: Int, by secondNumber: Int) -> Int {
+    return firstNumber * secondNumber
+}
+
+func digitsMultiplication(_ firstNumber: Double, by secondNumber: Double) -> Int {
+    return Int(firstNumber * secondNumber)
+}
+
+func digitsMultiplication(_ firstNumber: Int, by secondNumber: Int) -> Double {
+    return Double(firstNumber * secondNumber)
+}
+
+func digitsMultiplication(_ firstNumber: Double, by secondNumber: Double) -> Double {
+    return firstNumber * secondNumber
+}
+
+//Division implementation via overloading
+
+let theFirstDivide: Int     = digitsDivision(2, by: 2)
+let theSecondDivide: Double = digitsDivision(4, by: 8)
+let theThirdDivide: Int     = digitsDivision(5.0, by: 8.0)
+let theFourthDivide: Double = digitsDivision(5.5, by: 5.6)
+
+func digitsDivision(_ firstNumber: Int, by secondNumber: Int) -> Int {
+    return firstNumber / secondNumber
+}
+
+func digitsDivision(_ firstNumber: Double, by secondNumber: Double) -> Int {
+    return Int(firstNumber / secondNumber)
+}
+
+func digitsDivision(_ firstNumber: Int, by secondNumber: Int) -> Double {
+    return Double(firstNumber / secondNumber)
+}
+
+func digitsDivision(_ firstNumber: Double, by secondNumber: Double) -> Double {
+    return firstNumber / secondNumber
+}
+
+//For this homework , it seems to me that i can use such maning for variables as : theFirstDivide , theFirstMultiply and etc.
```

Comment:
Sorry, can you clarify this bit a little bit more? I can't quite understand

Reply:
The comment explained why i use  a, an and the in namings of variables 
In the future i'll try to avoid such cases 
Will fix all variables